File: src/main/java/com/github/alexandrecarlton/idea/settings/applier/editor/EditorSettingsApplier.java
package com.github.alexandrecarlton.idea.settings.applier.editor;

import com.github.alexandrecarlton.idea.settings.applier_api.SettingsApplier;
import com.github.alexandrecarlton.idea.settings.layout.editor.EditorSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.CodeStyleSettings;
import com.intellij.openapi.project.Project;

public class EditorSettingsApplier implements SettingsApplier<EditorSettings> {

  private final SettingsApplier<CodeStyleSettings> codeStyleSettingsApplier;

  public EditorSettingsApplier(SettingsApplier<CodeStyleSettings> codeStyleSettingsApplier) {
    this.codeStyleSettingsApplier = codeStyleSettingsApplier;
  }

  public void apply(Project project, EditorSettings configuration) {
    configuration.codeStyle().ifPresent(codeStyle -> codeStyleSettingsApplier.apply(project, codeStyle));
  }
}

File: applier/impl/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/editor/EditorSettingsApplier.kt
package com.github.alexandrecarlton.idea.settings.dialog.editor

import com.github.alexandrecarlton.idea.settings.dialog.SettingsApplier
import com.github.alexandrecarlton.idea.settings.dialog.editor.codestyle.CodeStyleSettings
import com.github.alexandrecarlton.idea.settings.dialog.editor.general.GeneralSettings
import com.github.alexandrecarlton.idea.settings.dialog.editor.inspections.InspectionsSettings
import com.github.alexandrecarlton.idea.settings.dialog.editor.spelling.SpellingSettings
import javax.inject.Inject

class EditorSettingsApplier @Inject
constructor(private val generalSettingsApplier: SettingsApplier<GeneralSettings>,
            private val codeStyleSettingsApplier: SettingsApplier<CodeStyleSettings>,
            private val inspectionsSettingsApplier: SettingsApplier<InspectionsSettings>,
            private val spellingSettingsApplier: SettingsApplier<SpellingSettings>) : SettingsApplier<EditorSettings> {

    override fun apply(settings: EditorSettings) {
        settings.general?.let(generalSettingsApplier::apply)
        settings.codeStyle?.let(codeStyleSettingsApplier::apply)
        settings.inspections?.let(inspectionsSettingsApplier::apply)
        settings.spelling?.let(spellingSettingsApplier::apply)
    }
}
